### PR TITLE
Remove deprecated snake_case args.

### PR DESCRIPTION
**Pull request by schmmd**

### CODE REVIEW

**Comment by schmmd** [comment]
@joelgrus @matt-gardner @DeNeutoy are we ready to remove these deprecated flags?  They exist and are deprecated in v0.3.0.

**Comment by matt-gardner** [comment]
It's fine with me.

File: tests/commands/predict_test.py [hunk 1]
```diff
@@ -26,31 +26,22 @@ def test_add_predict_subparser(self):
...
-                      "--output-file", "/dev/null",  # this one was always kebab-case
-                      "--batch_size", "10",
-                      "--cuda_device", "0",
-                      "--silent"]
-
         kebab_args = ["predict",          # command
                       "/path/to/archive", # archive
                       "/dev/null",        # input_file
-                      "--output-file", "/dev/null",
+                      "--output-file", "/dev/null",  # this one was always kebab-case
```
Comment (joelgrus) [inline tests/commands/predict_test.py]: I don't think you need this comment anymore

File: tests/commands/evaluate_test.py [hunk 2]
```diff
@@ -27,18 +27,13 @@ def setUp(self):
 
     @flaky
     def test_evaluate_from_args(self):
-        snake_args = ["evaluate", "tests/fixtures/bidaf/serialization/model.tar.gz",
-                      "--evaluation_data_file", "tests/fixtures/data/squad.json",
-                      "--cuda_device", "-1"]
-
         kebab_args = ["evaluate", "tests/fixtures/bidaf/serialization/model.tar.gz",
```
Comment (joelgrus) [inline tests/commands/evaluate_test.py]: I would rename this, maybe to `argv`

File: allennlp/commands/predict.py [hunk 3]
```diff
@@ -67,7 +67,7 @@ def __init__(self, predictor_overrides: Dict[str, str] = {}) -> None:
         self.predictors = {**DEFAULT_PREDICTORS, **predictor_overrides}
 
     def add_subparser(self, name: str, parser: argparse._SubParsersAction) -> argparse.ArgumentParser:
-        # pylint: disable=protected-access
+         # pylint: disable=protected-access
```
Comment (joelgrus) [inline allennlp/commands/predict.py]: why did you add a space here?

File: tests/commands/predict_test.py [hunk 4]
```diff
@@ -26,31 +26,22 @@ def test_add_predict_subparser(self):
...
 
-        snake_args = ["predict",          # command
-                      "/path/to/archive", # archive
-                      "/dev/null",        # input_file
-                      "--output-file", "/dev/null",  # this one was always kebab-case
-                      "--batch_size", "10",
-                      "--cuda_device", "0",
-                      "--silent"]
-
         kebab_args = ["predict",          # command
```
Comment (joelgrus) [inline tests/commands/predict_test.py]: I would also rename this to `argv`
